### ISSUE TITLE
Memory Leak: Pinia store with the initial state defined outside causes memory leak

### ISSUE THREAD

**Issue by nishant-compro** [nuxt/nuxt#29736]
### Environment
- Nuxt Version:	3.12.4
- Vue Version:		3.4.38
- Pinia Version:	2.2.2
- Node Version: 	>20.11.0
- Operating System:   Linux
- Nitro Version:    	2.9.7
- Package Manager:  pnpm@9.12.2

### Description
The issue is observed when using a Pinia store within a Nuxt middleware, specifically when initializing the store with its initial state value being defined in a variable outside of store definition. The issue does not occur when the store is initialized with values in place.

We are unable to find the root cause. It is coming in Vue v3.4.38 and not in v3.5.0. Was there any issue like this which may have been fixed in v3.5.0?

**Working Example**
```js
// stores/myStore.js

export const useMyStore = defineStore({
  id: 'myStore',
  state: () => ({ data: { abc: 123 } }),
  getters: {
    getData: (state) => state.data
  }
})


```

**Problematic Example**
```js
// stores/myStore.js
const data = { abc: 123 };


export const useMyStore = defineStore({
  id: 'myStore',
  state: () => ({
      data: data
  }),
  getters: {
    getData: (state) => state.data
  }
})

```

**Middleware**
```js
// middleware/example.js
export default defineNuxtRouteMiddleware(() => {
  const abc = useMyStore().getData
})
```

### Expected Behavior
- Memory Leak should not occur

### Actual Behavior
- Memory leak occurs, with duplicate string being created after server side rendering.

![Image](https://github.com/user-attachments/assets/377b52ed-ae42-4ed0-8904-35926e34d9a9)

### Additional context
- The issue appears to be specifically related to the interaction between middleware and initialized store state
- The memory leak does not occur when the store is used in regular components using the same store configuration

**Comment by saifobeidat** [comment]
We faced the same exact issue before..

However, Vue team refactored their reactivity system in v3.5 which introduces good performance optimization for SSR.. checkout these:

https://blog.vuejs.org/posts/vue-3-5#reactivity-system-optimizations
https://github.com/vuejs/core/pull/10397

**Comment by nishant-compro** [comment]
> We faced the same exact issue before...


Hi @saifobeidat, were you able to find out the actual cause of the issue in the previous version? If yes, can you please share? Thank you.

**Comment by RLatypov** [comment]
@nishant-compro 

try doing this

`// middleware/example.js
export default defineNuxtRouteMiddleware(() => {
  const myStore = useMyStore();
  const abc = myStore.getData;

  myStore.$dispose();
})`

it helped me

**Comment by danielroe** [comment]
closed as this is resolved in Vue 3.5+

**Comment by dhondtlaurens** [comment]
I had the same issue when using a `.map` to define a const outside the initial state. The problem was fixed by moving the `.map` logic directly inside the state definition of the store, tying it to the store’s lifecycle and preventing memory leaks.
